src/core_plugins/kibana/server/lib/management/saved_objects/relationships.js

```diff
@@ -0,0 +1,222 @@
+function formatId(id) {
+  return id.split(':')[1];
+}
+
+async function findDashboardRelationships(id, size, callCluster, savedObjectsClient) {
+  const kibanaIndex = savedObjectsClient._index;
+  const dashboard = await savedObjectsClient.get('dashboard', id);
+  const visualizations = [];
+
+  // TODO: should we handle exceptions here or at the parent level?
```

Comment:
This level understands ES exceptions, not the higher level. We could throw something more useful to the higher level. In particular `_mget` "errors" are special since they're document-by-document unless the whole request is rejected (`429`).